browser/ui/speedreader/speedreader_bubble_controller.h

```diff
@@ -0,0 +1,56 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_SPEEDREADER_SPEEDREADER_BUBBLE_CONTROLLER_H_
+#define BRAVE_BROWSER_UI_SPEEDREADER_SPEEDREADER_BUBBLE_CONTROLLER_H_
+
+#include "brave/browser/ui/views/speedreader/speedreader_bubble_global.h"
+#include "content/public/browser/web_contents_user_data.h"
+
+namespace content {
+class WebContents;
+}  // namespace content
+
+namespace speedreader {
+
+class SpeedreaderBubbleView;
+
+class SpeedreaderBubbleController
```

Comment:
We have a naming convention for this kind of controller - `FooTabHelper`.
and I have a question - This controller should be TabHelper?
It means each tab will have its own `SpeedreaderBubbleController` instance.

Reply:
Got it. Since there is already a SpeedreaderTabHelper, I will move that logic in there. And yeah, every tab needs one of these as the domain from the webcontents is needed. See the second picture https://github.com/brave/brave-core/pull/8978#issuecomment-851811056

Reply:
Will we move currrent speedreader button to location bar?
If so, I think button could be controller for this bubble?

Reply:
AFAIK, `SpeedreaderButton::Update` is called when browser ui should be updated such as active tab changing, current tab's navigation state changing and etc...
I think `SpeedreaderButton` would be a good controller for these new bubble?

```
void SpeedreaderButton::Update(content::WebContents* active_contents) {
  if (active_contents) {
    auto* tab_helper =
        speedreader::SpeedreaderTabHelper::FromWebContents(active_contents);
    if (tab_helper) {
      const bool active = tab_helper->IsActiveForMainFrame();
      if (active_ != active) {
        active_ = active;
        UpdateImageAndText();
      }
    }
  }
}
```

Reply:
Yeah, this is going to move to the location bar. That is just left out to keep this PR smaller. The design of this is similar to [QRCodeGeneratorBubbleController](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/qrcode_generator/qrcode_generator_bubble_controller.h?q=qrcodegenerator&ss=chromium%2Fchromium%2Fsrc). The bubble controller will be called inside the button. Having it separated out also makes testing a bit easier.

Reply:
Then, I think we don't need controller class. Instead, button could show bubble.

Reply:
If this bubble could be launched from multiple places, I think we need controller class.

Reply:
Ah, I see - new speedreader button will be `PageActionIconView`